lib/npg_pipeline/function/s3_archiver.pm

```diff
@@ -0,0 +1,265 @@
+package npg_pipeline::function::s3_archiver;
+
+use namespace::autoclean;
+
+use File::Basename;
+use File::Spec::Functions qw{catdir catfile};
+use Moose;
+use Readonly;
+use Try::Tiny;
+
+use npg_pipeline::function::definition;
+
+extends 'npg_pipeline::base';
+
+with 'npg_pipeline::base::config';
+
+Readonly::Scalar my $ARCHIVE_EXECUTABLE => 'aws';
+Readonly::Scalar my $ARCHIVE_CONFIG     => 's3_archive.json';
+
+Readonly::Scalar my $CONFIG_ITEMS_KEY   => 's3_archive';
+Readonly::Scalar my $CONFIG_STUDY_KEY   => 'study_id';
+Readonly::Scalar my $CONFIG_URL_KEY     => 'url';
+
+our $VERSION = '0';
+
+
+=head2 expected_files
+
+  Arg [1]    : Data product whose files to list,npg_pipeline::product.
+
+  Example    : my @files = $obj->expected_files($product)
+  Description: Return a list of the files expected to to present for
+               archiving in the runfolder.
+
+  Returntype : Array
+
+=cut
+
+sub expected_files {
+  my ($self, $product) = @_;
+
+  $product or $self->logconfess('A product argument is required');
+
+  my @expected_files;
+
+  my $dir_path = catdir($self->archive_path(), $product->dir_path());
+  my @extensions = qw{cram cram.md5 cram.crai
+                      seqchksum sha512primesums512.seqchksum
+                      F0x900.stats F0xB00.stats F0xF04_target.stats
```

Comment:
These should have the suffixes `_F0x900.stats` `_F0xB00.stats` and `_F0xF04_target.stats` (as we have the convention of using the `_` to indicate some sort of filtering/subsetting of the data), but bundled with these other suffixes it will look for `.F0x900.stats` `.F0xB00.stats` `.F0xF04_target.stats`.

Reply:
Done